protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -255,4 +263,61 @@ public T parseBytes(byte[] serialized) {
       }
     }
   }
+
+  private static final class KnownLengthByteBufferIterable implements Iterable<ByteBuffer> {
+    private final InputStream stream;
+    private final int length;
+
+    private KnownLengthByteBufferIterable(InputStream stream, int length) {
+      this.stream = stream;
+      this.length = length;
+      stream.mark(length);
```

Comment:
This is unused

Reply:
Obsolete, as we are making a list up-front.

(It was for avoiding throw when creating the first iterator.)

Reply:
Oh, for the reset(). Makes sense.